src/app/utils/sample-url-generation.ts

```diff
@@ -82,6 +82,33 @@ export function removeExtraSlashesFromUrl(url: string): string {
 
 export function hasWhiteSpace(url: string): boolean {
   const parts = url.split('?');
-  const whitespaceChars = [' ', '\t', '\n', '%20'];
-  return whitespaceChars.some((char) => parts[0].includes(char));
+  const whitespaceChars = [' ', '\t', '\n', '%20', '\r'];
+  const allParts = url.split(/(%20|\s)/);
+  return getWhiteSpace(parts, allParts, whitespaceChars);
+}
+
+export function getWhiteSpace(parts: string[], allParts: string[], whitespaceChars: string[]): boolean {
+  const urlHasArgs = parts.length > 1;
+  if (urlHasArgs) {
+    const hasWhiteSpaceBeforeArgs = whitespaceChars.some((char) => parts[0].trimStart().includes(char));
```

Comment:
Looks like trimming the start of the content might remove existing spaces and cause the URL to fail the whitespace test